### PR TITLE
[Feature] add undo prop to VAFileUpload

### LINKED ISSUE #1735: Add prop to make `VaFileUpload` undo action optional
When deleting a file from `VaFileUpload` there is an `undo` action:
fileUpload

- [ ] Add a `prop` to make this optional.

Additional info:
`Undo` action will show up only when `type` `prop` is set to `list` or `gallery`.
[Initial discord reply](https://discord.com/channels/457593300799717377/771405157527191585/971459220711034950).

### PR DESCRIPTION
**Pull request by eyuelberga**
## Description

close #1735

**Note** : 
After merging this commit 8449e357ce4948d76f048aef69aa7a6a140075a0 some components stopped working properly including `VAFileUpload`.  It is related to `shallowReadonly`. I have noticed this on `VaAccordion`, `VaCheckbox`, `VaCollapse`.

This is the warning that comes up on the console:

```
​Set operation on key "modelValue" failed: target is readonly. Proxy
set @ reactivity.esm-bundler.js:545
```
I didn't create an issue because the commit is in the develop branch

## Markup:

<details>

```vue
// Your code
```
</details>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

### CODE REVIEW

File: packages/ui/src/components/va-file-upload/VaFileUploadGalleryItem/VaFileUploadGalleryItem.vue [hunk 1]
```diff
@@ -4,10 +4,12 @@
     class="va-file-upload-gallery-item"
     :class="{'va-file-upload-gallery-item--undo': removed}"
   >
+  <template v-if="undo">
     <va-file-upload-undo
       class="va-file-upload-gallery-item--undo"
       @recover="recoverImage"
     />
+  </template>
```
Comment (RVitaly1978) [inline packages/ui/src/components/va-file-upload/VaFileUploadGalleryItem/VaFileUploadGalleryItem.vue:12]:
```suggestion
    <va-file-upload-undo
      v-if="undo"
      class="va-file-upload-gallery-item--undo"
      @recover="recoverImage"
    />
```

File: packages/ui/src/components/va-file-upload/VaFileUploadListItem/VaFileUploadListItem.vue [hunk 2]
```diff
@@ -1,16 +1,18 @@
...
     no-padding
     :class="{'file-upload-list-item--undo': removed}"
   >
+  <template v-if="removed">
     <va-file-upload-undo
       @recover="recoverFile"
-      v-if="removed"
+      v-if="undo"
     />
+  </template>
```
Comment (RVitaly1978) [inline packages/ui/src/components/va-file-upload/VaFileUploadListItem/VaFileUploadListItem.vue:15]:
```
<va-file-upload-undo
  v-if="removed && undo"
  @recover="recoverFile"
/>
```

**Review by RVitaly1978 — CHANGES_REQUESTED** [review]

**Review by RVitaly1978 — APPROVED** [review]

**Review by m0ksem — APPROVED** [review]
How about having undo button under X button? 
image


Also, would be nice to have undoTimeout prop. Or choose better name.
It is hard for me in 2sec to move touchpad to undo button 😂, count that I need to think 1.5sec.

P.S. you can see on paint screenshot I need to move touchpad 3 times to get "undo" button.

**Comment by eyuelberga** [comment]
I have added an `undoDuration` prop to adjust the undo timeout

**Review by RVitaly1978 — APPROVED** [review]

**Review by m0ksem — APPROVED** [review]
Thanks! 🎉

**Review by aluarius — APPROVED** [review]

**Pull request merged**